log10/secrets.py

```diff
@@ -0,0 +1,40 @@
+"""
+This module provides a class for managing secrets.
+A secret is a key value pair that comes from the enviornment 
+"""
+
+import os
+from dotenv import load_dotenv
+
+
+class SecretsManager:
```

Comment:
Hi @delip , greatly appreciate your PR. 
One comment: there's a `Log10Config` class defined [here](https://github.com/log10-io/log10/blob/4cb7a1c65aa35b267eeb900a582ed3ab681faeb0/log10/llm.py#L16). It would be better to combine `SecretsManager` and `Log10Config` (`Log10Config` being user facing API). It could be move the functions in `SecretsManager` to `Log10Config` or let `Log10Config` to directly call `SecretsManager`. 
Please let us know if you have bandwidth to make the change. Thanks.

Reply:
Hi @delip ,

Thanks for the PR!

Mind take a look at @wenzhe-log10 's suggestion? Since we don't have anyone using this pattern, would otherwise close for now

Reply:
I'm only seeing this right now. Sorry. @nqn, please feel free to prioritize this as you see fit. It's not impacting my work, but it's nice to have.